test/runtests.jl

```diff
@@ -601,3 +601,21 @@ end
     @test popfirst!(x) == "1"
     @test x == ["2"]
 end
+
+@testset "constructor corner cases" begin
+    x = Vector{Any}(undef, 3)
+    y = PooledArray(x)
+    @test y isa PooledArray{Any}
+    @test all(iszero, y.refs)
```

Comment:
Maybe also test directly the expected API?
```suggestion
    @test !any(isassigned, y)
    @test all(iszero, y.refs)
```

Reply:
OK - I have fixed your proposal to make it work.